internal/cli/import_yaml.go

```diff
@@ -0,0 +1,221 @@
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].AllowedOrigins[j] = str
+			}
+		}
+
+		for j, item := range client.GrantTypes {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].GrantTypes[j] = str
+			}
+		}
+
+	}
+}
+
+func (t *TenantConfig) replaceResourceServersConfig(key string, replacement interface{}) {
+	for i, rs := range t.ResourceServers {
+
+		if strings.ContainsAny(rs.Name, key) {
+			str := strings.ReplaceAll(rs.Name, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.ResourceServers[i].Name = str
+		}
+
+		if strings.ContainsAny(rs.Identifier, key) {
+			str := strings.ReplaceAll(rs.Identifier, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.ResourceServers[i].Identifier = str
+		}
+
+		if strings.ContainsAny(rs.SigningAlg, key) {
+			str := strings.ReplaceAll(rs.SigningAlg, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.ResourceServers[i].SigningAlg = str
+		}
+
+		if strings.ContainsAny(rs.TokenDialect, key) {
+			str := strings.ReplaceAll(rs.TokenDialect, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.ResourceServers[i].TokenDialect = str
+		}
+
+		if strings.ContainsAny(rs.SigningSecret, key) {
+			str := strings.ReplaceAll(rs.SigningSecret, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.ResourceServers[i].SigningSecret = str
+		}
+
+		for j, s := range rs.Scopes {
+			if strings.ContainsAny(s.Description, key) {
+				str := strings.ReplaceAll(s.Description, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.ResourceServers[i].Scopes[j].Description = str
+			}
+		}
+
+	}
+}
+
+func (t *TenantConfig) replaceRolesConfig(key string, replacement interface{}) {
+	for i, r := range t.Roles {
+		for j, p := range r.Permissions {
+			if strings.ContainsAny(p.ResourceServerIdentifier, key) {
+				str := strings.ReplaceAll(p.ResourceServerIdentifier, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Roles[i].Permissions[j].ResourceServerIdentifier = str
+			}
+		}
+	}
+}
+
+func (t *TenantConfig) CheckForDuplicateNames() error {
+	var result *multierror.Error
+
+	clientNames := make(map[string]bool)
+
+	for _, c := range t.Clients {
+		if clientNames[c.Name] {
+			errMsg := fmt.Sprintf("found duplicate name in client: %s", c.Name)
+			result = multierror.Append(result, errors.New(errMsg))
+		}
+		clientNames[c.Name] = true
+	}
+
+	resourceServerNames := make(map[string]bool)
+	for _, rs := range t.ResourceServers {
+		if resourceServerNames[rs.Name] {
+			errMsg := fmt.Sprintf("found duplicate name in resourceServer: %s", rs.Name)
+			result = multierror.Append(result, errors.New(errMsg))
+		}
+		resourceServerNames[rs.Name] = true
+	}
+
+	roleNames := make(map[string]bool)
+	for _, r := range t.Roles {
+		if roleNames[r.Name] {
+			errMsg := fmt.Sprintf("found duplicate name in resourceServer: %s", r.Name)
```

Comment:
Probably missed by me earlier, but should change to `found duplicate name in role`
```suggestion
			errMsg := fmt.Sprintf("found duplicate name in role: %s", r.Name)
```

Reply:
Thank you!